posts/2023-09-30-keycloak-with-openliberty.adoc

```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title: "Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jimmy1wu
+author_github: https://github.com/jimmy1wu
+seo-title: Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak - OpenLiberty.io
+seo-description: Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT.
+blog_description: "Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT."
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak
+Jimmy Wu <https://github.com/jimmy1wu>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT.
+In this blog post, a Keycloak server has been setup with a realm called `openliberty` which contains a client called `sample-openliberty-keycloak` and the realm roles of `admin` and `user`. Additionally in the `sample-openliberty-keycloak` client, client authentication has been enabled, `http://localhost:9090/Callback` has been added as a valid redirect URI, and the `microprofile-jwt` client scope has been set to `Default`.
```

Comment:
I think we're missing a bit of context here on why someone may want to use Keycloak and it's advantages. Could possibly also cover more on why integration with Jakarta EE and MP or even alternatives and why this is a good choice?
At the moment, it feels like we jump straight into a technical tutorial/documentation without context of why I'd want to read this or use it.

Reply:
Also feel like it jumps into a lot of specific technical jargon which should be explained (e.g. what is a realm?).

Reply:
Might also be nice to give a short sentence on what Jakarta security and what MicroProfile JWT are (maybe with links to our guides on these in case readers have not used them before).